File: 024.rb
TARGET = 1000000 - 1

def nth_permutation(digits, target)
  return digits[0] if digits.size == 1
  possibilities = (1...digits.size).to_a.inject(:*)

  i = 0
  loop do
    break if target >= (possibilities) * i && target < (possibilities) * (i+1)
    i += 1
  end

  target = target % ((possibilities) * i)
  return digits.slice!(i) + nth_permutation(digits, target)
end

puts nth_permutation(('0'..'9').to_a, TARGET)

File: 024.py
import functools
import operator

TARGET = 1000000 - 1

def nth_permutation(digits, target):
    if len(digits) == 1:
        return str(digits[0])
    possibilities = functools.reduce(
        operator.mul, list(range(1, len(digits)))
    )

    i = int(target // possibilities)
    target = target % (possibilities * i)
    digit = digits[i]
    digits.remove(digit)
    return str(digit) + nth_permutation(digits, target)

print(nth_permutation([x for x in range(0,10)], TARGET))
